### PR TITLE
Update cloud terraform docs for modular modules

### PR DESCRIPTION
**Pull request by GoodOldJack12**
Docs to accommodate changes in https://github.com/hpcugent/openstack-templates/pull/11

### CODE REVIEW

File: source/cloud/terraform.md [hunk 1]
```diff
@@ -103,141 +103,53 @@ file in a public place.
...
-But usually you do not want to deploy all these examples, you can just
-keep the required module and comment out the rest. You can do this from
-environment directory:
-
+### Single VM(s)
+This module deploys one VM with a public IP address and can be configured with extra options
+Copy the `single.tf.example` code to the `main.tf` file like so:
 ```shell
-cd ~/openstack-templates/terraform/environment
+cat single.tf >> main.tf
```
Comment (alvarosimon) [inline source/cloud/terraform.md:120]: `cat single.tf.example...`

**Review by alvarosimon — APPROVED** [review]

File: source/cloud/terraform.md [hunk 2]
```diff
@@ -103,141 +103,53 @@ file in a public place.
...
-previous step (see previous
-[section](#getting-terraform-examples)), go to the scripts directory:
-
-```shell
-cd ~/openstack-templates/terraform/scripts
-```
+We've provided two examples of how to use the terraform modules.
 
-And now run the script (usually you only have to run this script once).
+navigate to the environment directory first:
```
Comment (stdweird) [inline source/cloud/terraform.md:110]: start with capital

File: source/cloud/terraform.md [hunk 3]
```diff
@@ -103,141 +103,53 @@ file in a public place.
...
+----------------|-------------|----
+| vm_name       | Sets the name of the virtual machine. | (string)
+| image_name    | Sets the operating system image for the machine. | See [Image list](https://cloud.vscentrum.be/dashboard/project/images)
+| flavor_name   | Sets the machine flavor. | see [Flavors list](flavors.md).
+| nginx_enabled | Installs nginx and exposes port 80. | true, false
+| nfs_enabled   | Connects the vfm to the NFS network. Only set true if you requested access. | true, false
+| vsc_enabled   | Connects the vfm to the VSC network. Only set true if you requested access. | true, false
+
+More advanced options are described further on.
+### Cluster of VMs
```
Comment (stdweird) [inline source/cloud/terraform.md:134]: add extra newline

File: source/cloud/terraform.md [hunk 4]
```diff
@@ -103,141 +103,53 @@ file in a public place.
...
-But usually you do not want to deploy all these examples, you can just
-keep the required module and comment out the rest. You can do this from
-environment directory:
-
+### Single VM(s)
+This module deploys one VM with a public IP address and can be configured with extra options
+Copy the `single.tf.example` code to the `main.tf` file like so:
 ```shell
-cd ~/openstack-templates/terraform/environment
+cat single.tf.example >> main.tf
```
Comment (stdweird) [inline source/cloud/terraform.md:120]: why not jst copy?

File: source/cloud/terraform.md [hunk 5]
```diff
@@ -286,40 +215,104 @@ you a message after creating all the required resources.
...
+ SSH: ssh -A rocky@10.113.1.135
+MyCluster-private-1:
+ SSH: ssh -A rocky@10.113.1.229
+MyCluster-private-2:
+ SSH: ssh -A rocky@10.113.1.213
+
+EOT
+```
+:::{tip}
+To access your private VMs you need to ssh to your public VM first, then ssh from your main vm to your private VMs.
```
Comment (stdweird) [inline source/cloud/terraform.md:251]: try to stik to the same terminology `main vm` -> `public VM`

File: source/cloud/terraform.md [hunk 6]
```diff
@@ -286,40 +215,104 @@ you a message after creating all the required resources.
...
-It is possible to execute `./modify_variable.sh` script several times,
-(for instance, if you have changed the variables within
-`modify_variable.config` file). This will update your current Terraform
-deployment. Any Terraform variables change may affect to your current
-running VM(s) as well, `terrafrom apply` will inform you about this and
-it will request confirmation. The external ports for port forwarding
-are generated randomly when you run `modify_variable.sh` by default, but
-you can keep the current port forwarding rules if you add `-k` option:
-`./modify_variable.sh -k`
+Do not remove or modify the port_something_ssh.json files. These keep track of the external ports used. Deleting them _will_ break Terraform.
```
Comment (stdweird) [inline source/cloud/terraform.md:275]: quotes aroubd teh filename

File: source/cloud/terraform.md [hunk 7]
```diff
@@ -286,40 +215,104 @@ you a message after creating all the required resources.
...
+:::
+## Making changes to module variables
+### Adding NFS Or Nginx
```
Comment (stdweird) [inline source/cloud/terraform.md:278]: `Or` -> `or`

File: source/cloud/terraform.md [hunk 8]
```diff
@@ -286,40 +215,104 @@ you a message after creating all the required resources.
...
+:::{tip}
+If you enabled these options upon initial creation of the VM, you don't have to do these steps
 :::
 
+If you added NFS or Nginx after initial creation, you need to run a script **on your openstack VM** to mount the NFS volume or install nginx, respectively.
```
Comment (stdweird) [inline source/cloud/terraform.md:283]: `openstack VM` -> `VM`

File: source/cloud/terraform.md [hunk 9]
```diff
@@ -286,40 +215,104 @@ you a message after creating all the required resources.
...
+
+First ssh to your instance. You can get the command with
+```shell
+terraform output
+```
+Then run the command you got from terraform.
+After connecting to your instance, run:
+```shell
+curl http://169.254.169.254/2009-04-04/user-data | sudo bash
```
Comment (stdweird) [inline source/cloud/terraform.md:292]: this is an example or always this command?

File: source/cloud/terraform.md [hunk 10]
```diff
@@ -286,40 +215,104 @@ you a message after creating all the required resources.
...
+| access_key | the name of the ssh key you want to associate with the vm/cluster | (string)
+| playbook_url | A url to an ansible playbook that gets applied when nginx_enabled = true | (string) |
+| project_name | The name VSC of the project you want to create the resurce in | VSC_XXXX |
+| nfs_size | Size of the NFS share if nfs_enabled=true | (number) |
+
+#### Single VM only
+| Variable | Explanation | Values |
+| --- | --- | --- |
+| public | Add a public IP if true | true/false|
+#### Cluster only
```
Comment (stdweird) [inline source/cloud/terraform.md:307]: newline

File: source/cloud/terraform.md [hunk 11]
```diff
@@ -286,40 +215,104 @@ you a message after creating all the required resources.
...
+| Variable | Explanation | Values |
+| --- | --- | ---|
+| private_image | Sets a different OS name for the private VMs | See [Image list](https://cloud.vscentrum.be/dashboard/project/images) |
+| private_flavor | Sets a different flavor for the private VMs | see [Flavor list](https://cloud.vscentrum.be/dashboard/project/images) |
```
Comment (stdweird) [inline source/cloud/terraform.md:311]: wrong url to flavor

File: source/cloud/terraform.md (hunk 9 above)
Comment (GoodOldJack12) [reply source/cloud/terraform.md:292]: Surprisingly yeah its always this command

File: source/cloud/terraform.md (hunk 4 above)
Comment (stdweird) [reply source/cloud/terraform.md:120]: ok. maybe add a suggestion to use git (no need to explain it)

**Review by stdweird — APPROVED** [review]

**Review by stdweird — APPROVED** [review]

**Review by hajgato — APPROVED** [review]

**Pull request merged**